### DISCUSSION TITLE
Icons inside table cells are not vertically centered

### DISCUSSION THREAD

**Discussion by rcdailey** [mdx-js/mdx#2259] [Q&A] [answered]
First, I am not a frontend developer. I am using Docusaurus which I use for documentation pages and my understanding is they use MDX for parsing and rendering Markdown, so I'm coming here with this question assuming it's the right place. I apologize in advance if I am in the wrong place.

I have a markdown table that I render an icon inside of using Iconify:

```md
| Service     |                        Supported                        |
| ----------- | :-----------------------------------------------------: |
| Sonarr (v3) | <icon icon="fa6-solid:xmark" height="24" color="red" /> |
| Sonarr (v4) |             ![Yes](/icon/checkmark-16.png)              |
| Radarr      |             ![Yes](/icon/checkmark-16.png)              |
```

Ignore the last two rows. The first row (under the header row) is the one I'm playing with. When this renders it looks like this:

![image](https://user-images.githubusercontent.com/1768054/220153235-37cf3415-ed0e-4d7a-adda-ce06d021e868.png)

The red "X" isn't vertically centered. When I inspect the HTML for this specific cell, I see this:

```html
<td align="center">
```

![image](https://user-images.githubusercontent.com/1768054/220153498-052cea3f-2857-44dc-8b18-afcf511ebf8e.png)

If I change it to this, it looks much better:

```html
<td align="center" style="display: flex; align-items: center; justify-content: center;">
```

![image](https://user-images.githubusercontent.com/1768054/220157073-456d3bf0-36c7-4a5c-8a3c-2b5e42ba8d7b.png)

I can't tell if these styles are coming from MDX or Docusaurus. I just don't have the expertise to know where to look for the corresponding CSS. At the very least, I'm hoping someone here can help me find out. Otherwise, maybe there's a way to make table cells vertically center content so it looks better.

Thank you!

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @rcdailey! 👋 
Sorry you ran into a spot of trouble.

MDX + `remark-gfm` here is translating the GFM table to HTML following the GFM spec https://github.github.com/gfm/#tables-extension-
Neither the GFM spec, remark-gfm, nor MDX add styles.

Those are something which would be added from your project using a stylesheet.
You can apply your desired styles across the project with something like
```css
td[align=center] {
  display: flex;
  align-items: center;
  justify-content: center;
}
```

For how to add a global stylesheet, follow docusaurus' example here: https://docusaurus.io/docs/styling-layout/

**Reply by rcdailey** [reply]
Thank you so much for the quick reply! Upon trying your global CSS, it seems to have some negative impact on the cell borders. It looks like it's putting 1px borders adjacent to each other, which makes some borders appear to be 2px instead of 1 in width. Example:

![image](https://user-images.githubusercontent.com/1768054/220166732-34647456-0988-4a2f-8319-21a4732ebec9.png)

Also if you look at the very bottom of the table, you'll see that the border is misaligned as well.

**Reply by ChristianMurphy** [reply]
Yes, it may require additional styles to make the table look exactly how you want it to for your design system.
There are great resources which introduce working with CSS like:
* https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/CSS_basics
* https://www.w3schools.com/Css/
* https://web.dev/learn/css/

And the Docusaurus community may be able to provide some guidance on how to create a theme for their platform https://docusaurus.io/community/support
